modules/nw-running-dpdk-rootless-tap.adoc

```diff
@@ -0,0 +1,260 @@
+// Module included in the following assemblies:
+//
+// * networking/hardware_networks/using-dpdk-and-rdma.adoc
+
+:_content-type: PROCEDURE
+[id="nw-running-dpdk-rootless-tap_{context}"]
+= Running the DPDK workload with kernel access
+
+It is desirable to allow certain types of traffic for example log messages to take a exception path from your DPDK application. To support this you need access to the kernel networking and the tap device plugin provides this capability. This allows the DPDK application to run as a non privileged user (rootless).
+
+.DPDK and TAP example configuration
+image::rootless_dpdk_example.png[DPDK and TAP plugin]
+
+.Prerequisites
+
+* Install the OpenShift CLI (`oc`).
+* Install the SR-IOV Network Operator.
+* Log in as a user with `cluster-admin` privileges.
+* Run `setsebools container_use_devices=true` as root on all nodes.
```

Comment:
Can we provide an official way to activate this selinux boolean? May be MachineConfig resource example ?

Reply:
I added link in additional resources section maybe I need to reference here too

Reply:
This is the MachineConfig I applied 
```
apiVersion: machineconfiguration.openshift.io/v1
kind: MachineConfig
metadata:
  labels:
    machineconfiguration.openshift.io/role: worker
  name: 99-worker-setsebool
spec:
  config:
    ignition:
      version: 3.2.0
    systemd:
      units:
      - contents: |
          [Unit]
          Description=Set SELinux booleans
          Before=kubelet.service

          [Service]
          Type=oneshot
          ExecStart=/usr/sbin/setsebool container_use_devices true

          [Install]
          WantedBy=multi-user.target graphical.target
        enabled: true
        name: setsebool.service
```

However, I cannot see it applied when I log into the node though I think MachineConfig is correct 

```
Temporary namespace openshift-debug-w9hs6 is created for debugging node...
Starting pod/mysno-snodemolab-debug ...
To use host binaries, run `chroot /host`
Pod IP: 192.168.122.72
If you don't see a command prompt, try pressing enter.
sh-4.4# chroot /host
sh-5.1# getsebool container_use_devices
container_use_devices --> off
```

Reply:
This is a prerequisite which is always needed for the tap cni. Don't include it in the procedure. I completely agree with @kononovn. There should be an example, please create a new PR in the tap cni section with this. The official crd is:

apiVersion: machineconfiguration.openshift.io/v1
kind: MachineConfig
metadata:
  labels:
    machineconfiguration.openshift.io/role: worker
  name: 99-worker-setsebool
spec:
  config:
    ignition:
      version: 3.2.0
    systemd:
      units:
        - contents: |
            [Unit]
            Description=Set SELinux boolean for tap cni plugin
            Before=kubelet.service
            
            [Service]
            Type=oneshot
            ExecStart=/sbin/setsebool container_use_devices=on
            RemainAfterExit=true
            
            [Install]
            WantedBy=multi-user.target graphical.target
          enabled: true
          name: setsebool.service

Please add this in the documentation. Note that it takes a lot of time (15 minutes or so) to be applied. you can check it with oc get machineconfigpools.